docs/DeveloperGuide.md

```diff
@@ -5,9 +5,147 @@
+
+- Step 5: This value is set to `User.caloriesRequired` by `User#setCaloriesRequired(int)`.
+
+#### Design considerations
+
+- **Alternative 1 (current choice):** Uses an API to get the calories needed
+    - Pros: No need to figure out the optimal algorithm
+    - Cons: Need to parse response to sieve out necessary information
+
+- **Alternative 2:** Uses an algorithm to find the number of calories needed
+  - Pros: Not dependent on external APIs
+  - Cons: Need to come up with an algorithm to use
+
+### Calories list feature
+
+The `calories list` feature lists out the record of all the Calories data that the user has keyed in. The Calories data are all stored into a `ArrayList<Entry> calorieArrayList` attribute of the `CalorieList` Class. Calories data are printed when the `printCalorieList()` function is called. 
+
+The `printCalorieList()` function iterates through the `calorieArrayList` and prints out the Entries according to its order in the Array List.
+
+The Class diagram and sequence diagram for Calories list feature is shown below. Unrelated attributes and Classes were excluded.
+
+![CaloriesListClassDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/DevGuideRex/docs/CaloriesListClassDiagram.puml)
+![CaloriesListSequenceDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/RexDG/docs/CaloriesListSequenceDiagram.puml)
+
+### Calories delete feature
+
+The `calories delete` feature can delete the calories record at specific index of calorie list. This functionality is facilitated by `CaloriesList`. It implements one operation, namely:
+- `deleteEntry(String line)`
+
+Given below is an example usage scenario and how this mechanism behaves at every step:
+- Step 1: When the user inputs the command `calories delete INDEX` in the terminal, the string is sent to `Ui#handleUserInput()`, which will call `Ui#handleCaloriesInput()`.
+
+- Step 2: After the `Ui#handleCaloriesInput()` matching `delete calories` key word, the string will be passed into deleteEntry(String line) to execute delete process.
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The calories record (`Entry`) stored in the `ArrayList<Entry> caloriesList` will be deleted by calling `calorieArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `CalorieListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest calories list will be updated to saving file by calling `CalorieList#updateFile()`.
+
+The Class diagram for Calories delete feature is shown below:
+
+![CaloriesDeleteClassDiagram](https://github.com/a-wild-chocolate/tp/blob/master/docs/caloriesDeleteUML.jpg)
```

Comment:
The diagram does not seem to appear correctly in the Developer Guide. Would it be better to update the link to a relative path in the code, e.g. "caloriesDeleteUML.jpg" instead of the github link?